Crypto/Number/F2m.hs

```diff
@@ -209,3 +210,16 @@ divF2m
     -- ^ Quotient
 divF2m fx n1 n2 = mulF2m fx n1 <$> invF2m fx n2
 {-# INLINE divF2m #-}
+
+traceF2m :: BinaryPolynomial -> Integer -> Integer
+traceF2m fx = foldr addF2m 0 . take (log2 fx) . iterate (squareF2m fx)
+{-# INLINE traceF2m #-}
+
+halfTraceF2m :: BinaryPolynomial -> Integer -> Integer
+halfTraceF2m fx = foldr addF2m 0 . take (1 + log2 fx `div` 2) . iterate (squareF2m fx . squareF2m fx)
```

Comment:
I would suggest to use ``` `unsafeShiftR` 1``` instead of `div`.

Reply:
I can do that. I am assuming `(!>>.)` would also be okay?

Reply:
It is not provided by old versions, sigh.
So, I recommend `unsafeShiftR`.

Reply:
Ah, I see, that is unfortunate. I have changed it to use `unsafeShiftR`.